test/e2e/install_test.go

```diff
@@ -99,10 +109,137 @@ var _ = Describe("Operator Install", func() {
 
 		})
 		AfterEach(func() {
+			if basePath := env.GetString("ARTIFACT_PATH", ""); basePath != "" {
+				// get all the artifacts from the test run and save them to the artifact path
+				getArtifactsOutput(basePath)
+			}
 			err := c.Delete(ctx, operatorCatalog)
 			Expect(err).ToNot(HaveOccurred())
 			err = c.Delete(ctx, operator)
 			Expect(err).ToNot(HaveOccurred())
 		})
 	})
 })
+
+// getArtifactsOutput gets all the artifacts from the test run and saves them to the artifact path.
+// right now it will save:
+// - operators
+// - pods logs
+// - deployments
+// - bundle
+// - bundledeployments
+// - catalogsources
+
+func getArtifactsOutput(basePath string) {
+	ctx := context.Background()
+	cfg = ctrl.GetConfigOrDie()
+	kubeClient, err := kubeclient.NewForConfig(cfg)
+	Expect(err).To(Not(HaveOccurred()))
+
+	artifactPath := path.Join(basePath, fmt.Sprint(time.Now().UnixNano())) + "/"
```

Comment:
I think you want filepath.Join? Also, I don't think you should need to append the trailing slash?

Reply:
fixed